semgrep-core/parsing/Parse_kotlin_tree_sitter.ml

```diff
@@ -1432,41 +1465,47 @@ and primary_expression (env : env) (x : CST.primary_expression) : expr =
        let v4 = token env v4 (* ")" *) in
        let v5 =
          (match v5 with
-          | `Cont_stru_body x -> control_structure_body env x
-          | `SEMI tok -> token_todo env tok (* ";" *)
+          | `Cont_stru_body x ->
+              let v1 = control_structure_body env x in
+              (v1, None)
+          | `SEMI tok ->
+              token_todo env tok (* ";" *)
           | `Opt_cont_stru_body_opt_SEMI_else_choice_cont_stru_body (v1, v2, v3, v4) ->
               let v1 =
                 (match v1 with
                  | Some x -> control_structure_body env x
                  | None -> todo env ())
               in
-              let v2 =
+              (*let v2 =
                 (match v2 with
                  | Some tok -> token env tok (* ";" *)
                  | None -> todo env ())
-              in
+                in*)
               let v3 = token env v3 (* "else" *) in
               let v4 =
                 (match v4 with
                  | `Cont_stru_body x -> control_structure_body env x
                  | `SEMI tok -> token_todo env tok (* ";" *)
                 )
               in
-              todo env (v1, v2, v3, v4)
+              (v1, Some(v4))
          )
        in
-       todo env (v1, v2, v3, v4, v5)
+       let (v6, v7) = v5 in
+       let if_stmt = If (v1, v3, v6, v7) in
+       OtherExpr (OE_StmtExpr, [S if_stmt])
```

Comment:
maybe you can use Conditional instead, but it accepts 3 expressions.

Reply:
ah yes -- the 3 expressions part is difficult because v5 returns a statement 😢